spec/manager_spec.rb

```diff
@@ -1,9 +1,151 @@
 require "webmock/fixtures"
 
 describe WebMock::Fixtures::Manager do
-  describe "constructor" do
-    it "should not fail" do
-      WebMock::Fixtures::Manager.new()
+  after(:example) do
+    # Ensure we remove any added fixtures between tests
+    WebMock::Fixtures::Manager.reset!
+  end
+
+  before(:example) do
+    # Register a single fixture before each test
+    WebMock::Fixtures::Manager.register_fixture(:get_example, :get, %r{www.example.org}, :body => "Hello World")
+  end
+
+  describe "::reset!" do
+    it "should remove any registered fixtures" do
+      expect(WebMock::Fixtures::Manager.registered_fixtures[:get_example]).to_not(be_nil)
+      WebMock::Fixtures::Manager.reset!
+      expect(WebMock::Fixtures::Manager.registered_fixtures).to(eq({}))
+    end
+  end
+
+  describe "::register_fixture" do
+    context "with a subclassed manager" do
+      it "should not share registered fixtures" do
+        class TestManager < WebMock::Fixtures::Manager
+          @registered_fixtures = {
+            :get_google => {
+              :verb => :get,
+              :pattern => %r{www.google.com},
+              :response => {
+                :body => "Google",
+              },
+            },
+          }
+        end
+
+        # Ensure TestManager has the appropriate fixtures
+        expect(TestManager.registered_fixtures.keys()).to(eq([:get_google]))
+        expect(TestManager.registered_fixtures[:get_google]).to_not(be_nil)
+
+        # Ensure WebMock::Fixtures::Manager has the appropriate fixtures
+        expect(WebMock::Fixtures::Manager.registered_fixtures.keys()).to(eq([:get_example]))
+        expect(WebMock::Fixtures::Manager.registered_fixtures[:get_google]).to(be_nil)
+      end
+    end
+
+    it "should properly register the fixture" do
+      expected = {
+        :pattern => %r{www.example.org},
+        :verb => :get,
+        :response => {
+          :body => "Hello World",
+        },
+      }
+      expect(WebMock::Fixtures::Manager.registered_fixtures[:get_example]).to(eq(expected))
+    end
+  end
+
+  describe "::load_fixture_file" do
+    before(:example) do
+      fixture_path = File.join(File.expand_path(File.dirname(__FILE__)), "get_httpbin_200.raw")
+      WebMock::Fixtures::Manager.load_fixture_file(:get_httpbin, :get, %r{^http://httpbin\.org/get$}, fixture_path)
+    end
+
+    it "should properly register a fixture" do
+      fixture = WebMock::Fixtures::Manager.registered_fixtures[:get_httpbin]
+      expect(fixture[:verb]).to(eq(:get))
+      expect(fixture[:pattern]).to(eq(%r{^http://httpbin\.org/get$}))
+
+      # Assert that we loaded the raw HTTP request from the file as a string
+      expect(fixture[:response]).to(be_a(String))
+      # Assert we have response headers
+      expect(fixture[:response]).to(match(%r{Content-Type: application/json}))
+      # Assert we have part of the response body
+      expect(fixture[:response]).to(match(%r{"url": "http://httpbin\.org/get\?page=1"}))
```

Comment:
I am realizing that my concern about this test having response content was initially misplaced. I think I want a test verifying that making an HTTP request when a fixture is "running" actually gets the response from the endpoint. However, I see nothing making an HTTP call =/